traitsui/qt4/helper.py

```diff
@@ -173,12 +173,27 @@ def __init__(self, icon, slot):
 
         # Configure the button.
         self.setIcon(ico)
-        self.setMaximumSize(ico_sz, ico_sz)
         self.setFlat(True)
         self.setFocusPolicy(QtCore.Qt.NoFocus)
 
         self.clicked.connect(slot)
 
+    def sizeHint(self):
+        """ Reimplement sizeHint to return a recommended button size based on
+        the size of the icon.
+
+        Returns
+        -------
+        size : QtCore.QSize
+        """
+        # We want the button to have a size similar to the icon.
```

Comment:
I should call `ensurePolished()` here for good measure: https://doc.qt.io/qt-5/qwidget.html#ensurePolished
This was also called in the original implementation. https://code.woboq.org/qt5/qtbase/src/widgets/widgets/qpushbutton.cpp.html#_ZNK11QPushButton8sizeHintEv